pkg/build/api/v1/types.go

```diff
@@ -869,6 +869,14 @@ type BuildConfigSpec struct {
 
 	// CommonSpec is the desired build specification
 	CommonSpec `json:",inline" protobuf:"bytes,3,opt,name=commonSpec"`
+
+	// SuccessfulBuildsHistoryLimit is the number of old successful builds to retain.
+	// This field is a pointer to allow for differentiation between an explicit zero and not specified.
```

Comment:
Remove the pointer comment it and replace with user facing comment: "if nod specified, alll successful builds are reyained